### PR TITLE
Add federation flag to server-acl-init-job

### PR DESCRIPTION
**Pull request by kschoche**
Changes proposed in this PR:
- Add a new flag `-federation` to server-acl-init-job so so that we know there are multiple datacenters being federated even when there is no acl replication being configured, such as with HCP.

How I've tested this PR:
Acceptance tests pass
Unit tests

How I expect reviewers to test this PR:
Code review, possibly HCS pointing at a consul-k8s build from (https://github.com/hashicorp/consul-k8s/pull/531)

Checklist:
- [x] Bats tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ commands:
         type: string
       consul-k8s-image:
         type: string
-        default: "docker.mirror.hashicorp.services/hashicorpdev/consul-k8s:latest"
+        default: "kschoche/consul-k8s-dev-fed"
```
Comment (kschoche) [inline .circleci/config.yml:51]: Will change this  back to normal once we merge the consul-k8s PR.

**Review by thisisnotashwin — APPROVED** [review]
nice!

**Review by lkysow — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (lkysow) [reply .circleci/config.yml]: I've done this and pushed it since Kyle's away tomorrow.

File: test/unit/server-acl-init-job.bats [hunk 2]
```diff
@@ -1431,3 +1431,19 @@ load _helpers
       yq '.spec.template.spec.containers[0].command | any(contains("create-controller-token"))' | tee /dev/stderr)
   [ "${actual}" = "true" ]
 }
+
+#--------------------------------------------------------------------
+# global.federation.enabled
+@test "serverACLInit/Job: ensure federation is passed when federation is enabled" {
```
Comment (lkysow) [inline test/unit/server-acl-init-job.bats:1437]:
```suggestion
# global.federation.enabled

@test "serverACLInit/Job: ensure federation is passed when federation is enabled" {
```

**Comment by lkysow** [comment]
Tests will fail because they rely on k8s image but this is good to merge.

**Pull request merged**